### PR TITLE
knowledge: dynamic knowledgebase fix

### PR DESCRIPTION
**Pull request by jalavik**
* Updates the dynamic knowledgebase API to use the new search API.

Signed-off-by: Jan Aage Lavik <jan.age.lavik@cern.ch>

Depends on #3346 

@hachreak @jmartinm

### CODE REVIEW

File: invenio/modules/knowledge/api.py [hunk 1]
```diff
@@ -547,32 +547,34 @@ def get_kbd_values_by_def(confdict, searchwith=""):
...
-                                                           cc=collection)
-    if reclist:
-        return [val for (val, dummy) in
-                search_engine.get_most_popular_field_values(reclist, field)]
+            response = Query("{0}:{1}".format(field, fake_exp)).search(
+                collection=collection
+            )
+    if response.recids:
+        values = []
+        for rec in response.records():
```
Comment (jirikuncar) [inline invenio/modules/knowledge/api.py]: for readability can you use `record` instead of `rec`?

File: invenio/modules/knowledge/api.py [hunk 2]
```diff
@@ -547,32 +547,34 @@ def get_kbd_values_by_def(confdict, searchwith=""):
...
                 fake_exp = searchwith
-            reclist = search_engine.perform_request_search(f=field, p=fake_exp,
-                                                           cc=collection)
-    if reclist:
-        return [val for (val, dummy) in
-                search_engine.get_most_popular_field_values(reclist, field)]
+            response = Query("{0}:{1}".format(field, fake_exp)).search(
+                collection=collection
+            )
+    if response.recids:
```
Comment (jirikuncar) [inline invenio/modules/knowledge/api.py]: @jalavik can we just use `response.records()` as usage of `response.recids` is very heavy? We might want to return only `field` in search results and also we need to set correct pagination.

File: invenio/modules/knowledge/api.py (hunk 1 above)
Comment (jalavik) [inline invenio/modules/knowledge/api.py]: Sure!

File: invenio/modules/knowledge/api.py (hunk 2 above)
Comment (jalavik) [inline invenio/modules/knowledge/api.py]: Good point!

File: invenio/modules/knowledge/api.py [hunk 3]
```diff
@@ -547,33 +546,35 @@ def get_kbd_values_by_def(confdict, searchwith=""):
     collection = ""
     if 'collection' in confdict:
         collection = confdict['collection']
-    reclist = []  # return this
+    response = None
```
Comment (jirikuncar) [inline invenio/modules/knowledge/api.py]: not needed

File: invenio/modules/knowledge/api.py [hunk 4]
```diff
@@ -547,33 +546,35 @@ def get_kbd_values_by_def(confdict, searchwith=""):
...
-                                                           cc=collection)
-    if reclist:
-        return [val for (val, dummy) in
-                search_engine.get_most_popular_field_values(reclist, field)]
-    return []  # in case nothing worked
+            response = Query("{0}:{1}".format(field, fake_exp)).search(
+                collection=collection
+            )
+
+    response.body["size"] = 9999999
```
Comment (jirikuncar) [inline invenio/modules/knowledge/api.py]: can you add `# TODO wait for new search API for pagination`? Thanks

**Comment by jirikuncar** [comment]
@jalavik thanks. Please check my comments, rebase and we can :shipit:.

**Comment by jalavik** [comment]
@jirikuncar Thanks. Added fixes and reviewer.

**Pull request merged**